### PR TITLE
KFLUXINFRA-3694 Add CI to validate tekton-kueue CEL expressions

### PR DESCRIPTION
**Pull request by manish-jangra**
## Summary

- Add GitHub Actions workflow (`.github/workflows/test-tekton-kueue-config.yaml`) that runs on PRs touching `components/kueue/**`
- Add Python test script (`hack/test-tekton-kueue-config.py`) that validates tekton-kueue CEL expressions using the actual container via Podman

## Test scenarios

The test covers all boundary conditions of the CEL configuration:

| Test case | What it validates |
|-----------|------------------|
| `signing_pipeline` | Full match: rate-limited=true + group=signing-server gets resource annotation |
| `non_signing_pipeline` | Empty labels: no resource annotation added |
| `no_labels_pipeline` | No labels field: `has(labels)` check fails gracefully |
| `rate_limited_false` | rate-limited=false: value check rejects correctly |
| `partial_labels_pipeline` | Wrong group value: group check rejects correctly |

All tests also validate that `pipelines-queue` and `konflux-default` priority are unconditionally assigned.

## Test plan

- [x] Verified locally with Podman: signing pipeline gets correct annotations/labels
- [x] Verified locally: non-signing pipeline does NOT get resource annotation
- [x] Python syntax validated
- [x] Pre-commit hooks pass
- [ ] CI runs successfully on this PR (workflow self-triggers on `components/kueue/**` changes)

## Note

The `internal-production` config path will be available once #341 merges. Until then, the CI prerequisite check will correctly flag its absence.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: .github/workflows/test-tekton-kueue-config.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - 'hack/test-tekton-kueue-config.py'
+      - '.github/workflows/test-tekton-kueue-config.yaml'
+
+jobs:
+  test-tekton-kueue-config:
+    name: Test Tekton-Kueue CEL expressions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```
Comment (filariow) [inline .github/workflows/test-tekton-kueue-config.yaml]:
```suggestion
        uses: actions/checkout@v6
```

File: hack/test-tekton-kueue-config.py [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+    "internal-staging": {
+        "name": "Internal Staging",
+        "config_file": "components/kueue/internal-staging/tekton-kueue/config.yaml",
+        "kustomization_file": "components/kueue/internal-staging/tekton-kueue/kustomization.yaml"
+    },
+    "internal-production": {
+        "name": "Internal Production",
+        "config_file": "components/kueue/internal-production/tekton-kueue/config.yaml",
+        "kustomization_file": "components/kueue/internal-production/tekton-kueue/kustomization.yaml"
+    }
```
Comment (filariow) [inline hack/test-tekton-kueue-config.py:321]: we don't have internal prod

**Review by filariow — CHANGES_REQUESTED** [review]

File: .github/workflows/test-tekton-kueue-config.yaml (hunk 1 above)
Comment (manish-jangra) [reply .github/workflows/test-tekton-kueue-config.yaml]: fixed

File: hack/test-tekton-kueue-config.py (hunk 2 above)
Comment (manish-jangra) [reply hack/test-tekton-kueue-config.py:321]: It will skip now, it doesn't find it
